src/fides/api/util/consent_util.py

```diff
@@ -228,3 +247,281 @@ def get_or_create_fides_user_device_id_provided_identity(
+
+    created_privacy_notices: List[PrivacyNotice] = []
+    affected_regions: Set = set()
+
+    for privacy_notice in privacy_notice_schemas:
+        if should_escape:
+            # should_escape flag is for when we're creating a notice
+            # from a template. The content was already escaped in the
+            # template - we don't want to escape twice.
+            privacy_notice = transform_fields(  # type: ignore
+                transformation=escape,
+                model=privacy_notice,
+                fields=PRIVACY_NOTICE_ESCAPE_FIELDS,
+            )
+        created_privacy_notice = PrivacyNotice.create(
+            db=db,
+            data=privacy_notice.dict(exclude_unset=True),
+            check_name=False,
+        )
+        created_privacy_notices.append(created_privacy_notice)
+        affected_regions.update(created_privacy_notice.regions)
+
+    # After creating any new notices, make sure experiences exist to back all notices.
+    upsert_privacy_experiences_after_notice_update(
+        db, affected_regions=list(affected_regions)
+    )
+    return created_privacy_notices, affected_regions
+
+
+def validate_privacy_notice_dry_update(dry_update: PrivacyNotice) -> None:
+    """
+    Verify that a dry update of a PrivacyNotice satisfies the constraints
+    for creating a privacy notice.
+
+    This is done here instead of upfront because we need access to the current values
+    of the privacy notice in the database combined with the patch updates from the request
+    """
+    try:
+        PrivacyNoticeCreation.from_orm(dry_update)
+    except ValueError as exc:
+        raise HTTPException(
+            status_code=HTTP_422_UNPROCESSABLE_ENTITY,
+            # pylint: disable=no-member
+            detail=exc.errors(),  # type: ignore
+        )
+
+
+def prepare_privacy_notice_patches(
+    privacy_notice_updates: List[PrivacyNoticeWithId],
+    db: Session,
+    model: Union[Type[PrivacyNotice], Type[PrivacyNoticeTemplate]],
+) -> List[
+    Tuple[PrivacyNoticeWithId, Optional[Union[PrivacyNotice, PrivacyNoticeTemplate]]]
+]:
+    """
+    Prepares PrivacyNotice or Template upserts including performing data use
+    conflict validation on proposed changes.
+
+    For Privacy Notices, returns a list of tuples with the update data alongside the existing Privacy
+    db record that will be updated.
+
+    For PrivacyNoticeTemplates, returns a list of tuples with the data to be upserted, alongside
+    the existing PrivacyNoticeTemplate if it exists or None otherwise.
+
+    :param privacy_notice_updates: List of privacy notice schemas with ids: appropriate
+    for editing PrivacyNotices or upserting PrivacyNoticeTemplates
+    :param db: Session
+    :param model: one of PrivacyNotice or PrivacyNoticeTemplate
+    :return:
+    """
+    allow_create: bool = model == PrivacyNoticeTemplate
+    ignore_disabled: bool = model != PrivacyNoticeTemplate
```

Comment:
Some new flags to flip some behavior in this method depending on whether the resource is a template or a notice